### PR TITLE
Support for HK format v1

### PR DESCRIPTION
**Pull request by mhasself**
Updates loaders, checkers, and testing to be compatible with current HK schema (v0) and a new schema (v1) that makes use of G3TimesampleMap.

### CODE REVIEW

**Comment by jlashner** [comment]
Just checked this with a new data file from here: https://github.com/simonsobs/ocs/pull/128 and it works. Have you checked the translator with old data?

File: python/hk/translator.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    def __init__(self):
+        self.stats = {'n_hk': 0,
+                      'n_other': 0,
+                      'versions': {}}
+
+    def Process(self, f):
+        """Translates one frame to the target schema.  Irrelevant frames are
+        passed through unmodified.
+
```
Comment (BrianJKoopman) [inline python/hk/translator.py:31]: Missing "Args:" and "Returns:" sections in docstring.

Comment (BrianJKoopman) [reply python/hk/translator.py:31]: Part of what motivated this comment was seeing all the returns here return a list `[f]`, and where `Translator` gets called it typically has a `[0]` index immediately following. I'm probably missing something, but why returns lists all the time here?

File: python/hk/scanner.py [hunk 2]
```diff
@@ -29,6 +31,7 @@ def __init__(self):
                 'n_error': 0,
                 'n_warning': 0
             },
+            'versions': {}, # Number of frames counted for each hkagg_version.
```
[the remark this answers is not in the document]
Comment (mhasself) [reply python/hk/scanner.py:34]: Yes, for sure... fixing in a new commit.

File: python/hk/translator.py (hunk 1 above)
Comment (mhasself) [reply python/hk/translator.py:31]: Docstring will be updated, good call.

The HKTranslator and HKScanner are both G3Pipeline-compatible, which means the Process/__call__ can return either a G3Frame object or a (possibly 0-len) list of frames.  But when using these things in non-G3Pipeline, it's better not to have those output types mixed.  The list is more general so I seem to have defaulted to it.  (I think I'm back-filling justification here... it's possible I was just confused about G3Modules for a while... but I think returning a list here is looking forward.)  Anyway, adding text.

File: python/hk/scanner.py [hunk 3]
```diff
@@ -19,6 +19,9 @@ class HKScanner:
           and error (key ``n_error``) events encountered.  The detail
           for such events is logged to ``spt3g.core.log_warning`` /
           ``log_error``.
+        - ``versions`` (dict): The number of frames (value) (value)
```
Comment (BrianJKoopman) [inline python/hk/scanner.py:22]: (value) (value)

**Pull request merged**